process/sync/shardForkDetector.go

```diff
@@ -82,8 +82,8 @@ func (sfd *shardForkDetector) AddHeader(
 func (sfd *shardForkDetector) addFinalHeaders(finalHeaders []data.HeaderHandler, finalHeadersHashes [][]byte) {
 	finalCheckpointWasSet := false
 	for i := 0; i < len(finalHeaders); i++ {
-		isFinalHeaderNonceHigherThanGenesis := finalHeaders[i].GetNonce() > process.GenesisBlockNonce
-		if isFinalHeaderNonceHigherThanGenesis {
+		isFinalHeaderNonceNotLowerThanCurrent := finalHeaders[i].GetNonce() >= sfd.finalCheckpoint().nonce
```

Comment:
Can you please explain why this was needed? And maybe add a unit test to cover this change?

Reply:
This was an old check but it was incomplete. As the finalHeader should always increase because these headers come from meta blocks and in the normal cases they would always be higher than the last ones. But one problem could appear in a roll back situation, when the final header could be set with a lower value despite the fact that this action would not be permitted.

Reply:
I added in Jira task EN-4194 for an integration test for this case

Reply:
Added unit test